src/plugins/orangehrmClaimPlugin/Controller/AssignClaimRequestController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Claim\Controller;
+
+use OrangeHRM\Claim\Traits\Service\ClaimServiceTrait;
+use OrangeHRM\Core\Authorization\Controller\CapableViewController;
+use OrangeHRM\Core\Controller\AbstractVueController;
+use OrangeHRM\Core\Controller\Common\NoRecordsFoundController;
+use OrangeHRM\Core\Controller\Exception\RequestForwardableException;
+use OrangeHRM\Core\Traits\Service\ConfigServiceTrait;
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Core\Vue\Component;
+use OrangeHRM\Core\Vue\Prop;
+use OrangeHRM\Entity\ClaimRequest;
+use OrangeHRM\Framework\Http\Request;
+
+class AssignClaimRequestController extends AbstractVueController implements CapableViewController
+{
+    use ClaimServiceTrait;
+    use UserRoleManagerTrait;
+    use ConfigServiceTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function preRender(Request $request): void
+    {
+        $id = $request->attributes->getInt('id');
+        $claimRequest = $this->getClaimService()->getClaimDao()->getClaimRequestById($id);
+        $empNumber = $claimRequest->getEmployee()->getEmpNumber();
+        $isPastEmployee = $claimRequest->getEmployee()->getEmployeeTerminationRecord() !== null;
+        $empName = $isPastEmployee ?
+            $claimRequest->getEmployee()->getFirstName() . ' ' . $claimRequest->getEmployee()->getLastName() . ' (Past Employee)'
+            : $claimRequest->getEmployee()->getFirstName() . ' ' . $claimRequest->getEmployee()->getLastName();
+
+        if ($this->getUserRoleManagerHelper()->isSelfByEmpNumber($claimRequest->getEmployee()->getEmpNumber())) {
+            $component = new Component('submit-claim');
+        } else {
+            $component = new Component('assign-claim');
```

Comment:
use redirection rather that switching components.
e.g. src/plugins/orangehrmAttendancePlugin/Controller/PunchInController.php
```php
            $this->setResponse($this->redirect('/attendance/punchOut'));
            return;
```